orchagent/orchdaemon.cpp

```diff
@@ -329,6 +330,26 @@ void OrchDaemon::start()
          * is a good chance to flush the pipeline before next select happened.
          */
         flush();
+
+        /*
+         * Asked to check warm restart readiness.
+         * Not doing this under Select::TIMEOUT condition because of
+         * the existence of finer granularity ExecutableTimer with select
+         */
+        if (gSwitchOrch->checkRestartReady())
+        {
+            bool ret = warmRestartCheck();
+            if (ret)
+            {
+                // Orchagent is ready to perform warm restart, stop processing any new db data.
+                // Should sleep here or continue handling timers and etc.??
+                if (!gSwitchOrch->checkRestartNoFreeze())
+                {
+                    SWSS_LOG_WARN("Orchagent is frozen for warm restart!");
+                    sleep(UINT_MAX);
```

Comment:
>sleep(UINT_MAX); [](start = 20, length = 16)

Can we just exit with some error code?

Reply:
Exiting orchagent may trigger unnecessary supervisord warning.  Keeping sleep() behavior to avoid that.

Reply:
It is easy to add supervisor config with some expected return code. I am asking whether this is the right thing to do.

Reply:
What is the problem for orchagent to sleep() and wait for official shutdown signal with other processes?
I don't see the need to perform supervisor change here, which is not necessary.